packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -956,9 +964,13 @@ export class ConnectionManager implements IConnectionManager {
 		await waitForOnline();
 
 		this.triggerConnect(
-			error !== undefined
-				? "Reconnecting due to Error"
-				: `Reconnecting due to: ${disconnectMessage}`,
+			{
+				description:
+					reason.error !== undefined
+						? "Reconnecting due to Error"
+						: `Reconnecting due to: ${reason.description}`,
+				error: reason.error,
```

Comment:
this is a place we were for sure losing an error, there are a few other suspicious spots, but frankly, the code is hard to follow, so many layers with the same methods names, and lots of passing between them. Hence the reason i went with this approach, as it was easier to ensure the right pieces get passed along.